File: datasets/python_source-code/Multiple-B1/exact_copy_iterative_1.py
# Binary Tree Traversal - unique source, ITERATIVE family
# This is the second of two "true source" files for this test case. It
# should match exact_copy_iterative_1.py, but must NOT cross-match with
# the recursive family (unique_recursive_1.py / exact_copy_recursive_1.py
# / renamed_reordered_combo_1.py), even though both solve the same
# in-order traversal problem.

class Node:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None


def insert(root, value):
    if root is None:
        return Node(value)
    if value < root.value:
        root.left = insert(root.left, value)
    else:
        root.right = insert(root.right, value)
    return root


def inorder_iterative(root):
    result = []
    stack = []
    current = root
    while current is not None or stack:
        while current is not None:
            stack.append(current)
            current = current.left
        current = stack.pop()
        result.append(current.value)
        current = current.right
    return result


def build_tree(values):
    root = None
    for v in values:
        root = insert(root, v)
    return root


if __name__ == "__main__":
    values = [50, 30, 70, 20, 40, 60, 80]
    root = build_tree(values)
    result = inorder_iterative(root)
    print("In-order traversal (iterative):", result)

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_b1/exact_copy_iterative_1.java
// Binary Tree Traversal - unique source, ITERATIVE family
// This is the second of two "true source" files for this test case.

import java.util.ArrayList;
import java.util.List;
import java.util.Stack;

class Node {
    int value;
    Node left;
    Node right;

    public Node(int value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

class Solution {
    public static Node insert(Node root, int value) {
        if (root == null) {
            return new Node(value);
        }
        if (value < root.value) {
            root.left = insert(root.left, value);
        } else {
            root.right = insert(root.right, value);
        }
        return root;
    }

    public static List<Integer> inorderIterative(Node root) {
        List<Integer> result = new ArrayList<>();
        Stack<Node> stack = new Stack<>();
        Node current = root;
        while (current != null || !stack.isEmpty()) {
            while (current != null) {
                stack.push(current);
                current = current.left;
            }
            current = stack.pop();
            result.add(current.value);
            current = current.right;
        }
        return result;
    }

    public static Node buildTree(int[] values) {
        Node root = null;
        for (int v : values) {
            root = insert(root, v);
        }
        return root;
    }

    public static void main(String[] args) {
        int[] values = {50, 30, 70, 20, 40, 60, 80};
        Node root = buildTree(values);
        List<Integer> result = inorderIterative(root);
        System.out.println("In-order traversal (iterative): " + result);
    }
}
